src/modules/hip/kernel/phase.hpp

```diff
@@ -0,0 +1,276 @@
+
+        rpp_hip_load8_and_unpack_to_float8(srcPtr1 + srcIdx, &src1_f8);
+        rpp_hip_load8_and_unpack_to_float8(srcPtr2 + srcIdx, &src2_f8);
+        phase_hip_compute(srcPtr1, &src1_f8, &src2_f8, &dst_f8, &multiplier_f1);
+        rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+    }
+}
+
+template <typename T>
+__global__ void phase_pkd3_pln3_hip_tensor(T *srcPtr1,
+                                           T *srcPtr2,
+                                           uint2 srcStridesNH,
+                                           T *dstPtr,
+                                           uint3 dstStridesNCH,
+                                           RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + ((id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    uint dstIdx = (id_z * dstStridesNCH.x) + (id_y * dstStridesNCH.z) + id_x;
+
+    float multiplier_f1 = (float)(255 / 1.570796);
+
+    d_float24 src1_f24, src2_f24, dst_f24;
+
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr1 + srcIdx, &src1_f24);
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr2 + srcIdx, &src2_f24);
+    phase_hip_compute(srcPtr1, &src1_f24.f8[0], &src2_f24.f8[0], &dst_f24.f8[0], &multiplier_f1);
+    phase_hip_compute(srcPtr1, &src1_f24.f8[1], &src2_f24.f8[1], &dst_f24.f8[1], &multiplier_f1);
+    phase_hip_compute(srcPtr1, &src1_f24.f8[2], &src2_f24.f8[2], &dst_f24.f8[2], &multiplier_f1);
+    rpp_hip_pack_float24_pln3_and_store24_pln3(dstPtr + dstIdx, dstStridesNCH.y, &dst_f24);
+}
+
+template <typename T>
+__global__ void phase_pln3_pkd3_hip_tensor(T *srcPtr1,
+                                           T *srcPtr2,
+                                           uint3 srcStridesNCH,
+                                           T *dstPtr,
+                                           uint2 dstStridesNH,
+                                           RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    float multiplier_f1 = (float)(255 / 1.570796);
```

Comment:
You have added a comment saying the constant computation has been moved outside, but it is still showing in the PR. Please verify once

Reply:
Yep this commit wasn't pushed into remote, now it has been